### PR TITLE
Fix NPE in sanitizeMetricName when unit or aggregationTemporality is …

### LINKED ISSUE #6683: Prometheus sink NPE in sanitizeMetricName when unit or aggregationTemporality is null.
**Describe the bug**
The prometheus sink throws NullPointerException in PrometheusTimeSeries.sanitizeMetricName() when processing metric events where the unit field or the aggregationTemporality field is null. The method calls .equals() on getAggregationTemporality() and .startsWith() on unit without null checks. These fields are optional in the OTEL metric model and are null when metrics come from sources that do not carry this metadata, such as Prometheus Remote Write v1.

**To Reproduce**
 1. Configure a Data Prepper pipeline with the prometheus source receiving Remote Write v1 data
 2. Configure the prometheus sink as the output of that pipeline
 3. Start a Prometheus instance with remote_write pointing to the Data Prepper source
 4. Observe repeated NullPointerException warnings in Data Prepper logs for every metric event

**Expected behavior**
The prometheus sink should handle null unit and null aggregationTemporality gracefully instead of throwing NullPointerException. A simple nullcheck before calling .equals() and .startsWith() on these fields would resolve both issues.

**Exceptions**
Stack trace for aggregationTemporality:
  NullPointerException: Cannot invoke "String.equals(Object)" because the return value of
  "org.opensearch.dataprepper.model.metric.Sum.getAggregationTemporality()" is null
      at PrometheusTimeSeries.sanitizeMetricName(PrometheusTimeSeries.java:305)
      at PrometheusTimeSeries.<init>(PrometheusTimeSeries.java:96)
      at PrometheusSinkBufferEntry.getTimeSeriesForEvent(PrometheusSinkBufferEntry.java:49)

  Stack trace for unit:
  NullPointerException: Cannot invoke "String.startsWith(String)" because "unit" is null
      at PrometheusTimeSeries.sanitizeMetricName(PrometheusTimeSeries.java:310)
      at PrometheusTimeSeries.<init>(PrometheusTimeSeries.java:96)
      at PrometheusSinkBufferEntry.getTimeSeriesForEvent(PrometheusSinkBufferEntry.java:49)

**Environment (please complete the following information):**
- OS: macOS (Darwin 25.3.0)
- Version: Data Prepper 2.15.0-SNAPSHOT (main branch)

**Additional context**
Found during end-to-end testing of the prometheus source plugin (PR #6627). The prometheus source produces valid OTEL metric events but Prometheus Remote Write v1 protocol does not carry unit or aggregation temporality metadata, so these fields are null. 

The sink was built assuming these fields are always populated, which is only true for metrics coming from the OTel Collector. Current workaround is to set sanitize_names: false in the prometheus sink configuration to bypass the sanitizeMetricName() method entirely.

### PR DESCRIPTION
**Pull request by srikanthpadakanti**
### Description
  Fix NPE in `PrometheusTimeSeries.sanitizeMetricName()` when `unit` or `aggregationTemporality` is null.
                                                                                                                                                    
  ### Issues Resolved
  Resolves #6683                                                                                                                                    
                                                            
  ### Check List
  - [x] New functionality includes testing.
  - [ ] New functionality has a documentation issue. Please link to it in this PR.
    - [ ] New functionality has javadoc added
  - [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/service/PrometheusTimeSeries.java [hunk 1]
```diff
@@ -302,38 +302,40 @@ static String sanitizeMetricName(final Metric metric) {
         final boolean isGauge = metric.getKind().equals(Metric.KIND.GAUGE.toString());
         final boolean isCounter = metric.getKind().equals(Metric.KIND.SUM.toString()) &&
                 ((Sum) metric).isMonotonic() &&
-                ((Sum) metric).getAggregationTemporality().equals("AGGREGATION_TEMPORALITY_CUMULATIVE");
+                "AGGREGATION_TEMPORALITY_CUMULATIVE".equals(((Sum) metric).getAggregationTemporality());
```
Comment (dlvenable) [inline data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/service/PrometheusTimeSeries.java:305]: Let's include a test case for when this is true and the unit is `null`.

```
@Test
public void testSumMetricCounterWithNullUnit() {
    final String name = RandomStringUtils.randomAlphabetic(10);
    Sum sum = createSumMetric(name, null, true, "AGGREGATION_TEMPORALITY_CUMULATIVE");
    String sanitizedName = PrometheusTimeSeries.sanitizeMetricName(sum);
    assertThat(sanitizedName, equalTo(name + "_total"));
}
```

Comment (srikanthpadakanti) [reply data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/service/PrometheusTimeSeries.java:305]: Sure.

File: data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/service/PrometheusTimeSeries.java [hunk 2]
```diff
@@ -302,38 +302,40 @@ static String sanitizeMetricName(final Metric metric) {
...
 
-        if (unit.startsWith("{")) {
+        if (unit != null && unit.startsWith("{")) {
             return metricNameBuilder.append(suffix).toString();
         }
 
-        if ("1".equals(unit) && isGauge) {
-            return metricNameBuilder.append(RATIO_SUFFIX).toString();
-        }
+        if (unit != null) {
```
Comment (kkondaka) [inline data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/service/PrometheusTimeSeries.java]: I think the right fix is to check for `unit` being null at the very beginning. Almost everything depends on unit

Comment (dlvenable) [reply data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/service/PrometheusTimeSeries.java]: What should be the behavior when `unit` is `null`? Do we not sanitize the name? Do we drop the event to DLQ?

Comment (srikanthpadakanti) [reply data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/service/PrometheusTimeSeries.java]: @kkondaka  sure. 

@dlvenable When unit is null we return the sanitized metric name with the counter suffix (if applicable), skipping all unit-based suffix logic. This is pretty muchsame  as the unit.startsWith("{") case.
